split_to_CMYK.jsx

```diff
@@ -1 +1,74 @@
var blackChannel = doc.activeChannels = [doc.channels.getByName('Black')];
*/

/*var CMYK = doc.changeMode(ChangeMode.CMYK);
var RGB = doc.changeMode(ChangeMode.RGB);

var addLayer = doc.artLayers.add();

var renameLayer = doc.artLayers.getByName("Layer 1")
renameLayer.name = "Cyan"

alert(renameLayer)*/



function makeLayers() {
    try {
        if (app.documents.length) {
            var layerColor = ["Cyan","Magenta","Yellow","Black"]
            var doc = activeDocument;
            var layers = doc.artLayers;
            var cmyk = [layerA= layers.add(),
                               layerB= layers.add(),
                               layerC= layers.add(),
                               layerD= layers.add()];

            if (layerColor) {
                cmyk[0].name = layerColor [3];
                cmyk[0].blendMode = BlendMode.MULTIPLY
                cmyk[1].name = layerColor [2];
                cmyk[1].blendMode = BlendMode.MULTIPLY
                cmyk[2].name = layerColor [1];
                cmyk[2].blendMode = BlendMode.MULTIPLY
                cmyk[3].name = layerColor [0];
                cmyk[3].blendMode = BlendMode.MULTIPLY
              
            }
            
            doc.activeLayer = newLayer;
        }
      } catch (e) { 
    }
}
makeLayers ()

/*function splitCyan () {
	var cyan = doc.activeChannels = [doc.channels.getByName('Cyan')];
	app.executeMenuCommand('selectall');
    
}
splitCyan()*/

/*var cyanChannel = doc.activeChannels = [doc.channels.getByName('Cyan')];
var magChannel = doc.activeChannels = [doc.channels.getByName('Magenta')];
var yelChannel = doc.activeChannels = [doc.channels.getByName('Yellow')];
var blackChannel = doc.activeChannels = [doc.channels.getByName('Black')];*/
\ No newline at end of file
+#target photoshop
+#targetengine "main"
+
+var doc = app.activeDocument;
+
+//switch color mode to get 4 channels
+doc.changeMode(ChangeMode.CMYK);
+
+//copies all channels into their own layers and adjusts blending mode
+//reverse order of var layerK - layerC so layers appear in proper order in PS
+function makeLayers() {
+		var layers = doc.artLayers;
+	
+		var layerK = layers.add();
+  			layerK.name = 'Black';
+  			layerK.blendMode = BlendMode.NORMAL;
+	
+		var layerY = layers.add();
+  			layerY.name = 'Yellow';
+  			layerY.blendMode = BlendMode.MULTIPLY;
+	
+		var layerM = layers.add();
+  			layerM.name = 'Magenta';
+  			layerM.blendMode = BlendMode.MULTIPLY;
+	
+		var layerC = layers.add();
+  			layerC.name = 'Cyan';
+  			layerC.blendMode = BlendMode.MULTIPLY;
+}
+makeLayers();
+
+//copies channels into corresponding layers
+function splitChannels(targetChannel) {
+    
+    doc.activeLayer = doc.artLayers.getByName("Background");
+
+    doc.activeChannels = [doc.channels.getByName(targetChannel)];
+        function selectCopy() {
+          doc.selection.selectAll();
+          doc.selection.copy();
+  }selectCopy();
+      
+      switch (targetChannel) {
```

Comment:
Since you're calling this function for each layer, you can skip the switch / case step!  Try this instead of the whole switch tree:

```
doc.activeLayer = doc.artLayers.getByName(targetChannel);
doc.paste();
```

2 lines should be able to do the job of 21!